datanode/snapshot/service.go

```diff
@@ -0,0 +1,186 @@
+package snapshot
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"code.vegaprotocol.io/vega/datanode/entities"
+	"code.vegaprotocol.io/vega/datanode/service"
+	"code.vegaprotocol.io/vega/datanode/sqlstore"
+	"code.vegaprotocol.io/vega/logging"
+)
+
+type Service struct {
+	log           *logging.Logger
+	config        Config
+	connConfig    sqlstore.ConnectionConfig
+	snapshotsPath string
+	blockStore    *sqlstore.Blocks
+	chainService  *service.Chain
+}
+
+func NewSnapshotService(log *logging.Logger, config Config, blockStore *sqlstore.Blocks, chainService *service.Chain, connConfig sqlstore.ConnectionConfig,
+	snapshotsPath string,
+) (*Service, error) {
+	service := &Service{
+		log:           log.Named("snapshot"),
+		config:        config,
+		connConfig:    connConfig,
+		snapshotsPath: snapshotsPath,
+		blockStore:    blockStore,
+		chainService:  chainService,
+	}
+
+	if service.snapshotsEnabled() {
+		_, err := os.Stat(service.snapshotsPath)
```

Comment:
Shouldn't we be looking in `config.DatabaseSnapshotsPath` if its set? Same when we in the `if config.RemoveSnapshotsOnStartup {}` statement.

Reply:
the database snapshot path is different from the nodes snapshot path (which is created using the Paths) and will depend on where and how the database is run as it is a path relative to the DB, and hence in the config, defaulted to the most usual case.